Assets/AnkrSDK/Scripts/Core/Utils/EventFilterHelper.cs

```diff
@@ -6,35 +6,18 @@ namespace AnkrSDK.Core.Utils
 {
 	public static class EventFilterHelper
 	{
-		public static NewFilterInput CreateEventFilters(EventBase eventHandler, EventFilterData evFilter = null)
+		public static NewFilterInput CreateEventFilters<TEvDto>(string contractAddress, EventFilterData evFilter = null)
 		{
-			if (evFilter == null)
-			{
-				return eventHandler.CreateFilterInput();
-			}
-
-			var isFilterTopic1Filled = evFilter.filterTopic1 != null;
-			var isFilterTopic2Filled = evFilter.filterTopic2 != null;
-			var isFilterTopic3Filled = evFilter.filterTopic3 != null;
+			var eventABI = ABITypedRegistry.GetEvent(typeof(TEvDto));
 			
-			if (isFilterTopic3Filled && isFilterTopic2Filled && isFilterTopic1Filled)
-			{
-				return eventHandler.CreateFilterInput(evFilter.filterTopic1, evFilter.filterTopic2,
-					evFilter.filterTopic3, evFilter.fromBlock, evFilter.toBlock);
-			}
-
-			if (isFilterTopic2Filled && isFilterTopic1Filled)
+			var ethFilterInput = FilterInputBuilder.GetDefaultFilterInput(contractAddress, evFilter.fromBlock, evFilter.toBlock);
+			if (evFilter.filterTopic1 != null || evFilter.filterTopic2 != null || evFilter.filterTopic3 != null)
```

Comment:
check with string.IsNullOrEmpty

Reply:
also move this check inside EventFilterData class so it will be like 
evFilter.IsFilled or something

Reply:
> check with string.IsNullOrEmpty

It is not a string it's a object[]